### ISSUE TITLE
Inspector: Create agents with hole from shapefile

### ISSUE THREAD

**Issue by benoitgaudou** [gama-platform/gama#1301]
Open and run the attached model (.zip.txt file, please remove the .txt extension to get a .zip archive file).

When we inspect a building with a hole, when we click on the shape, an exception is thrown (see below).

I have investigate a little deeper: 
when the agent has a geometry with a hole and we want to display its shape, the minus operator is called on 2 geometries ( the external one and the hole): in Spatial.java, line 1052 : return g1.difference(g2).

I made a little test on validity on the geometries : 
java.lang.System.out.println("g1 : " + g1.isValid() + " -- g2: " + g2.isValid());
I get : "g1 : false -- g2: true"

I appears that g1 (the external geometry) is not a valid geometry in the sense of JTS ...

Benoit

com.vividsolutions.jts.geom.TopologyException: side location conflict [ (920.2893782374449, 1072.6103248586878, 0.0) ]
	at com.vividsolutions.jts.geomgraph.EdgeEndStar.propagateSideLabels(EdgeEndStar.java:300)
	at com.vividsolutions.jts.geomgraph.EdgeEndStar.computeLabelling(EdgeEndStar.java:136)
	at com.vividsolutions.jts.geomgraph.DirectedEdgeStar.computeLabelling(DirectedEdgeStar.java:127)
	at com.vividsolutions.jts.operation.overlay.OverlayOp.computeLabelling(OverlayOp.java:373)
	at com.vividsolutions.jts.operation.overlay.OverlayOp.computeOverlay(OverlayOp.java:173)
	at com.vividsolutions.jts.operation.overlay.OverlayOp.getResultGeometry(OverlayOp.java:127)
	at com.vividsolutions.jts.operation.overlay.OverlayOp.overlayOp(OverlayOp.java:66)
	at com.vividsolutions.jts.operation.overlay.snap.SnapIfNeededOverlayOp.getResultGeometry(SnapIfNeededOverlayOp.java:96)
	at com.vividsolutions.jts.operation.overlay.snap.SnapIfNeededOverlayOp.overlayOp(SnapIfNeededOverlayOp.java:58)
	at com.vividsolutions.jts.geom.Geometry.difference(Geometry.java:1423)
	at msi.gaml.operators.Spatial$Operators.difference(Spatial.java:1053)
	at msi.gaml.operators.Spatial$Operators.minus(Spatial.java:1018)
	at gaml.additions.GamlAdditions$654.run(GamlAdditions.java:703)
	at gaml.additions.GamlAdditions$654.run(GamlAdditions.java:1)
	at msi.gaml.expressions.BinaryOperator.value(BinaryOperator.java:76)
	at msi.gama.runtime.AbstractScope.evaluate(AbstractScope.java:467)
	at msi.gama.util.GAML.evaluateExpression(GAML.java:75)
	at msi.gama.gui.parameters.ExpressionControl.computeValue(ExpressionControl.java:145)
	at msi.gama.gui.parameters.ExpressionControl.focusGained(ExpressionControl.java:178)
....

[Zombies.zip.txt](https://github.com/gama-platform/gama/files/33175/Zombies.zip.txt)

**Comment by ptaillandier** [comment]
The problem is due to the fact that GAMA tries to cast the string (shape value in the inspector) to a geometry. The question is: should we allow the user to modify a geometry (through a string) in the inspector? If no, we can easily solve this issue :)

**Comment by AlexisDrogoul** [comment]
Yep, the problem is due to the serialization / reinterpretation of the string in the inspector. 

However: 
1) I fixed the issue by catching the TopologyException (which was not caught previously) and doing a difference between the two geometries with a zero-buffer.

2) I noticed that the "holes" always return polylines (the GAML equivalent to LineString) instead of polygons, because that is what getInteriorRing() returns. I wonder if we shouldn't return polygons instead (that's probably what the user want to get, anyway).

3) By doing so, the geometry is "reinterpreted" and … it solves the problem exposed in Issue #1302. The "corrected" buildings have colors.

So this issue raises a question which is more or less: shouldn't we apply this 'fix' (i.e. forcing the serialization/deserialization) to all geometries with holes in them ? (and maybe explore point 2 as well, which may also help in computing these geometries).

Waiting for your advice, guys !
